src/marqo/core/index_management/index_management.py

```diff
@@ -452,6 +458,20 @@ def _add_default_query_profile(self, app: str) -> bool:
             return True
         return False
 
+    def _add_custom_searchers(self, app: str) -> bool:
+        target_path = os.path.join(app, 'components', self._MARQO_CUSTOM_SEARCHERS_JAR)
+        if not os.path.exists(target_path):
+            logger.debug('Custom searchers JAR does not exist. Copying from Vespa target directory')
+            current_dir = Path(__file__).parent
+            source_path = current_dir / f'../../../../vespa/target/{self._MARQO_CUSTOM_SEARCHERS_JAR}'
+            os.makedirs(os.path.dirname(target_path), exist_ok=True)
+            shutil.copy(source_path, target_path)
+
+            self._add_custom_searchers_to_services(app)
+
+            return True
+        return False
```

Comment:
Not necessary for this release. But if we need to roll out bugfix of searcher jar file, we'll need to compare the current one with the existing one (a md5 hash would be good enough), if different, copy over. we always copy the current one to override the existing one.